### PR TITLE
Create checkPantry method and pass test

### PR DESCRIPTION
**Pull request by aliroemhildt**
#### What's this PR do?
Create checkPantry method and pass test

#### Where should the reviewer start?
pantry.js and pantry-test.js

#### How should this be manually tested?
npm test

### CODE REVIEW

File: src/classes/Pantry.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+  checkPantry(recipe) {
+    let hasAllIngredients = true;
+
+    recipe.ingredients.forEach(recipeIng => {
+      const ingInPantry = this.ingredients.find(pantryIng => {
+        return recipeIng.id === pantryIng.ingredient;
+      })
+
+      if ((ingInPantry && (ingInPantry.amount < recipeIng.quantity.amount)) || !ingInPantry) {
```
Comment (anewb87) [inline src/classes/Pantry.js:14]: The magic of parens. We had the logic here but had to keep re-working till we realized we were missing a set of parens. Yippee!

File: src/data/sampleUser.js [hunk 2]
```diff
@@ -121,7 +121,7 @@ const sampleUser = [
       },
       {
         "ingredient": 19206,
-        "amount": 2
+        "amount": 3
```
Comment (anewb87) [inline src/data/sampleUser.js:124]: An oops that we fixed.

**Review by cagallo — APPROVED** [review]
This looks great, super clean checkPantry method!

**Review by anewb87 — APPROVED** [review]
Looks good! One more method to finish this out.

**Review by kammon10 — APPROVED** [review]
good catch!

**Pull request merged**